pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,577 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networkoverhead
+
+import (
+	"context"
+	"fmt"
+	agv1alpha1 "github.com/diktyo-io/appgroup-api/pkg/apis/appgroup/v1alpha1"
+	aglisters "github.com/diktyo-io/appgroup-api/pkg/generated/listers/appgroup/v1alpha1"
+	ntv1alpha1 "github.com/diktyo-io/networktopology-api/pkg/apis/networktopology/v1alpha1"
+	ntlisters "github.com/diktyo-io/networktopology-api/pkg/generated/listers/networktopology/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	"math"
+	pluginconfig "sigs.k8s.io/scheduler-plugins/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/networkaware/util"
+	networkawareutil "sigs.k8s.io/scheduler-plugins/pkg/networkaware/util"
+	"sort"
+)
+
+const (
+	// Name : name of plugin used in the plugin registry and configurations.
+	Name = "NetworkOverhead"
+
+	// MaxCost : MaxCost used in the NetworkTopology for costs between origins and destinations
+	MaxCost = 100
+
+	// SameHostname : If pods belong to the same host, then consider cost as 0
+	SameHostname = 0
+
+	// SameZone : If pods belong to hosts in the same zone, then consider cost as 1
+	SameZone = 1
+)
+
+// NetworkOverhead : Filter and Score nodes based on Pod's AppGroup requirements: MaxNetworkCosts requirements among Pods with dependencies
+type NetworkOverhead struct {
+	handle      framework.Handle
+	podLister   corelisters.PodLister
+	agLister    *aglisters.AppGroupLister
+	ntLister    *ntlisters.NetworkTopologyLister
```

Comment:
AppGroupLister and NetworkTopologyLister are interfaces, usually we don't apply another `*` to them:

```
agLister    aglisters.AppGroupLister
ntLister    ntlisters.NetworkTopologyLister
```